### DISCUSSION TITLE
Distinctions between some similarly named static methods

### DISCUSSION THREAD

**Discussion by ev-dev** [lzybkr/PSReadLine#3276] [Q&A] [answered]
I recently started making more advanced usage of this module by scripting some custom key handlers, and I'm trying to understand the intended purpose/usage of a few static methods that control cursor actions.

The methods I'm unclear about are those like `NextWord` vs. `ForwardWord` and also the distinction between methods containing 'Shell' in their name like the `NextWord` and `ShellNextWord` methods. All 4 of these methods seem to have different intended functionality, as their definitions are all unique within Movement.cs, but I'm not familiar enough to understand what their different purposes are.

Any insights are appreciated!

**Comment by lzybkr** [comment, marked as answer]
The public static methods should all be documented in the source code with XML comments. Alternatively, they should also be documented here: https://docs.microsoft.com/en-us/powershell/module/psreadline/about/about_psreadline?view=powershell-7.2

If that documentation is not clear enough, you can open a PR to improve the documentation or open an issue in the appropriate repo.
